src/igvfd/types/file_set.py

```diff
@@ -317,6 +318,30 @@ def protocols(self, request, input_file_sets=None):
                     protocols.update(protocol)
         return list(protocols)
 
+    @calculated_property(
+        schema={
+            'title': 'Functional Assay Mechanisms',
+            'description': 'The biological mechanism(s) the functional assay measures.',
+            'type': 'array',
+            'minItems': 1,
+            'uniqueItems': True,
+            'items': {
+                'title': 'Phenotype Term',
+                'type': 'string',
+                'linkTo': 'PhenotypeTerm'
+            },
+            'notSubmittable': True
+        }
+    )
+    def functional_assay_mechanisms(self, request, input_file_sets=None):
+        '''Calculate an array of unique biological mechanism(s) measured by the funcional assays associated with an analysis set.'''
```

Comment:
I like to have comments in the code but for IGVFd and encodeD we have avoided them usually :)

Reply:
I will remove it